examples/pytorch/bgnn/BGNN.py

```diff
@@ -0,0 +1,356 @@
+            elif self.task == 'classification':
+                metrics['loss'] = F.cross_entropy(pred, y.long())
+                metrics['accuracy'] = torch.Tensor([(y == pred.max(1)[1]).sum().item()/y.shape[0]])
+
+            return metrics
+
+
+    def train_and_evaluate(self, model_in, target_labels, train_mask, val_mask, test_mask,
+                           optimizer, metrics, gnn_passes_per_epoch):
+        loss = None
+
+        for _ in range(gnn_passes_per_epoch):
+            loss = self.train_model(model_in, target_labels, train_mask, optimizer)
+
+        self.model.eval()
+        logits = self.model(*model_in).squeeze()
+        train_results = self.evaluate_model(logits, target_labels, train_mask)
+        val_results = self.evaluate_model(logits, target_labels, val_mask)
+        test_results = self.evaluate_model(logits, target_labels, test_mask)
+        for metric_name in train_results:
+            metrics[metric_name].append((train_results[metric_name].detach().item(),
+                               val_results[metric_name].detach().item(),
+                               test_results[metric_name].detach().item()
+                               ))
+        return loss
+
+    def update_early_stopping(self, metrics, epoch, best_metric, best_val_epoch, epochs_since_last_best_metric, metric_name,
+                              lower_better=False):
+        train_metric, val_metric, test_metric = metrics[metric_name][-1]
+        if (lower_better and val_metric < best_metric[1]) or (not lower_better and val_metric > best_metric[1]):
+            best_metric = metrics[metric_name][-1]
+            best_val_epoch = epoch
+            epochs_since_last_best_metric = 0
+        else:
+            epochs_since_last_best_metric += 1
+        return best_metric, best_val_epoch, epochs_since_last_best_metric
+
+    def log_epoch(self, pbar, metrics, epoch, loss, epoch_time, logging_epochs, metric_name='loss'):
+        train_metric, val_metric, test_metric = metrics[metric_name][-1]
+        if epoch and epoch % logging_epochs == 0:
+            pbar.set_description(
+                "Epoch {:05d} | Loss {:.3f} | Loss {:.3f}/{:.3f}/{:.3f} | Time {:.4f}".format(epoch, loss,
+                                                                                              train_metric,
+                                                                                              val_metric,
+                                                                                              test_metric,
+                                                                                              epoch_time))
+
+    def fit(self, graph, X, y,
+            train_mask, val_mask, test_mask,
+            cat_features = None,
+            num_epochs=100,
+            patience=10,
+            logging_epochs=1,
+            metric_name='loss',
+            normalize_features=False,
+            replace_na=True,
+            ):
+        '''
```

Comment:
Same. Pls follow the numpy docstring style.  DGL also has some conventions for docing the argument types:
* Use `dgl.DGLGraph` for DGLGraph.
* Use `numpy.array` for numpy arrays and describe the shape in the description.
* Use `list[int]` for list of integers
* Use python type names for python objects (e.g., `int`, `float`, ...)